File: languages/js/src/entity/authenticity/hash_alg.ts
import * as proto from "../../bridge/proto/authenticity_entities";

/**
 * Represents different hash algorithms.
 */
export enum HashAlg {
  /**
   * Represents an unrecognized hash algorithm.
   */
  UNRECOGNIZED = -1,
  /**
   * Represents the SHA-256 hash algorithm.
   */
  Sha256 = 0,
  /**
   * Represents the Keccak-256 hash algorithm.
   */
  Keccak256 = 1,
  /**
   * Represents the Poseidon hash algorithm.
   */
  Poseidon = 2,
  /**
   * Represents no hash algorithm.
   */
  None = 3
}

export namespace HashAlg {
  export function fromProto(alg: proto.HashAlg): HashAlg {
    switch (alg) {
      case proto.HashAlg.SHA_256:
        return HashAlg.Sha256;
      case proto.HashAlg.KECCAK_256:
        return HashAlg.Keccak256;
      case proto.HashAlg.POSEIDON:
        return HashAlg.Poseidon;
      case proto.HashAlg.NONE:
        return HashAlg.None;
      default:
        return HashAlg.UNRECOGNIZED;
    }
  }

  export function toProto(alg: HashAlg): proto.HashAlg {
    switch (alg) {
      case HashAlg.Sha256:
        return proto.HashAlg.SHA_256;
      case HashAlg.Keccak256:
        return proto.HashAlg.KECCAK_256;
      case HashAlg.Poseidon:
        return proto.HashAlg.POSEIDON;
      case HashAlg.None:
        return proto.HashAlg.NONE;
      default:
        return proto.HashAlg.UNRECOGNIZED;
    }
  }
}

File: languages/go/entity/authenticity/hash_alg.go
package authenticity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// HashAlg represents different hash algorithms.
type HashAlg int32

// Constants representing specific hash algorithms.
const (
	// Sha256 represents the SHA-256 hash algorithm.
	Sha256       HashAlg = iota
	// Keccak256 represents the Keccak-256 hash algorithm.
	Keccak256    HashAlg = iota
	// Poseidon represents the Poseidon hash algorithm.
	Poseidon     HashAlg = iota
	// None represents no hash algorithm.
	None         HashAlg = iota
	// Unrecognized represents an unrecognized hash algorithm.
	Unrecognized HashAlg = -1
)

var (
	HashAlgFromProto = map[proto.HashAlg]HashAlg{
		proto.HashAlg_SHA_256:    Sha256,
		proto.HashAlg_KECCAK_256: Keccak256,
		proto.HashAlg_POSEIDON:   Poseidon,
		proto.HashAlg_NONE:       None,
	}

	HashAlgToProto = map[HashAlg]*proto.HashAlg{
		Sha256:    proto.HashAlg_SHA_256.Enum(),
		Keccak256: proto.HashAlg_KECCAK_256.Enum(),
		Poseidon:  proto.HashAlg_POSEIDON.Enum(),
		None:      proto.HashAlg_NONE.Enum(),
	}
)
